src/themis/secure_keygen.c

```diff
@@ -158,3 +167,21 @@ themis_status_t themis_is_valid_asym_key(const uint8_t* key, size_t length)
 
     return THEMIS_INVALID_PARAMETER;
 }
+
+themis_status_t themis_gen_sym_key(uint8_t* key, size_t* key_length)
+{
+    if (key_length == NULL) {
+        return THEMIS_INVALID_PARAMETER;
+    }
+    if (key != NULL && *key_length == 0) {
+        return THEMIS_INVALID_PARAMETER;
```

Comment:
shouldn't this be "buffer too small" as well?

Reply:
> shouldn't this be "buffer too small" as well?

Mmm... Yeah, it probably should.

My reasoning was that requesting zero-length key is an error, but it can be equally understood as "zero-length buffer is too small, please expand".

I've checked asymmetric key generation API and it returns THEMIS_BUFFER_TOO_SMALL when given non-NULL buffer and zero length. It also outputs proper length into the provided pointer (like when buffer is NULL). So I guess we'll use the same behavior:

```diff
-    if (key != NULL && *key_length == 0) {
-        return THEMIS_INVALID_PARAMETER;
-    }
-
-    if (key == NULL) {
+    if (key == NULL || *key_length == 0) {
         *key_length = THEMIS_SYM_KEY_LENGTH;
         return THEMIS_BUFFER_TOO_SMALL;
     }
```

Thanks for noticing!